gateway/gateway.go

```diff
@@ -15,7 +15,62 @@ import (
 
 // Config is the configuration used when creating a new gateway handler.
 type Config struct {
+	// Headers is a map containing all the headers that should be sent by default
+	// in all requests. You can define custom headers, as well as add the recommended
+	// headers via AddAccessControlHeaders.
 	Headers map[string][]string
+
+	// TrustedMode configures this gateway to allow trusted requests. By default,
+	// the gateway will operate in trustless mode, as defined in the specification:
+	// https://specs.ipfs.tech/http-gateways/trustless-gateway/.
+	//
+	// This only applies to hostnames not defined under PublicGateways. In addition,
+	// the hostnames localhost, 127.0.0.1 and ::1 are considered trusted by default.
```

Comment:
:warning:  :thought_balloon: I wonder if this is the right place to make this call.
Implicit defaults like that feel like a surface for security problems, when someone just uses library, then exposes it over 127.0.0.1 and then exposes that via Nginx to the public internet.

User did nothing wrong, just used default state of library, and exposed themselves to hosting phishing / bad content in deserialized form.

I think we should change this, so the default state for `boxo/gateway`  library is trustless, even on localhost.

@hacdias  mind refactoring things, so by default library is trustless-only, and we explicitly enable trusted mode on localhost names in examples? :pray: